pkg/cluster/cluster.go

```diff
@@ -1032,11 +1037,18 @@ func (c *Cluster) processPodEvent(obj interface{}) error {
 		return fmt.Errorf("could not cast to PodEvent")
 	}
 
+	podName := spec.NamespacedName(event.PodName)
+
 	c.podSubscribersMu.RLock()
-	subscriber, ok := c.podSubscribers[spec.NamespacedName(event.PodName)]
+	subscriber, ok := c.podSubscribers[podName]
 	c.podSubscribersMu.RUnlock()
 	if ok {
-		subscriber <- event
+		select {
+		case <-subscriber.stopEvent:
+			c.unregisterPodSubscriber(podName)
+		default:
```

Comment:
This is likely not a good default, it breaks the select semantics, you go into blocking wait for input to my understanding